Y/yaml_cpp/build_tarballs.jl

```diff
@@ -32,11 +40,12 @@ platforms = expand_cxxstring_abis(supported_platforms(; experimental=true))
 
 # The products that we will ensure are always built
 products = [
-    LibraryProduct("libyaml-cpp", :libyaml_cpp)
+    LibraryProduct(["libyaml-cpp", "yaml-cpp"], :libyaml_cpp)
 ]
 
 # Dependencies that must be installed before this package can be built
-dependencies = Dependency[
+dependencies = [
+    Dependency("boost_jll", v"1.79.0"),
```

Comment:
boost requires a compat with exactly a single version: #3024

Reply:
Fixed.

Not completely sure whether a BuildDependency would be enough here, but there is no warnings or errors from audit on x86_64-linux-gnu-cxx11 with just:
```julia
    BuildDependency(PackageSpec("boost_jll", v"1.76.0")),
```

Reply:
Ah, if you don't link to any boost library then a build-dep is fine, some components are headers-only.

Reply:
Right, judging from CMakeLists.txt, it also looks like there is no linking with boost...

... and Debian agrees that boost is a build-time dependency: https://salsa.debian.org/debian/yaml-cpp/-/blob/56d05874c8068235cc904370c209e459914ff1e8/debian/control#L9

Reply:
You can see if the output library links to boost (I'm away from computer, I can't)

Reply:
Changed to being a BuildDependency - with the same version (v1.76.0).

Reply:
Pending: Will check if it links to boost (but I guess audit would’ve caught that?)

Reply:
> (but I guess audit would’ve caught that?)

Not really, because the library _is_ there during audit.

Reply:
OK - it looks `boost`-free: 
```
/workspaces/Yggdrasil/Y/yaml_cpp# julia --project=../../.ci --color=yes build_tarballs.jl --verbose --debug=end x86_64-linux-gnu-cxx11
...

sandbox:${WORKSPACE}/srcdir/yaml-cpp/build # ls -la $libdir/libyaml-cpp*
lrwxrwxrwx    1 root     root            18 Oct 31 17:43 /workspace/destdir/lib/libyaml-cpp.so -> libyaml-cpp.so.0.5
lrwxrwxrwx    1 root     root            20 Oct 31 17:43 /workspace/destdir/lib/libyaml-cpp.so.0.5 -> libyaml-cpp.so.0.5.3
-rwxr-xr-x    1 root     root        571696 Oct 31 17:43 /workspace/destdir/lib/libyaml-cpp.so.0.5.3

sandbox:${WORKSPACE}/srcdir/yaml-cpp/build # ldd $libdir/libyaml-cpp.so.0.5.3
        linux-vdso.so.1 (0x00007ffe23a9f000)
        libstdc++.so.6 => /usr/lib/csl-glibc-x86_64/libstdc++.so.6 (0x00007f6c76f02000)
        libm.so.6 => /lib64/libm.so.6 (0x00007f6c76e28000)
        libgcc_s.so.1 => /usr/lib/csl-glibc-x86_64/libgcc_s.so.1 (0x00007f6c76e0d000)
        libc.so.6 => /lib64/libc.so.6 (0x00007f6c76c11000)
        ldd (0x00007f6c77393000)
```
 🚀